exchanges/okx/okx_wrapper.go

```diff
@@ -0,0 +1,1134 @@
+		Exchange:       ok.Name,
+		OrderID:        orderDetail.OrderID,
+		ClientOrderID:  orderDetail.ClientSupplierOrderID,
+		Side:           orderDetail.Side,
+		Type:           orderType,
+		Pair:           pair,
+		Cost:           orderDetail.Price,
+		AssetType:      assetType,
+		Status:         status,
+		Price:          orderDetail.Price,
+		ExecutedAmount: orderDetail.RebateAmount,
+		Date:           orderDetail.CreationTime,
+		LastUpdated:    orderDetail.UpdateTime,
+	}, err
+}
+
+// GetDepositAddress returns a deposit address for a specified currency
+func (ok *Okx) GetDepositAddress(ctx context.Context, c currency.Code, accountID, chain string) (*deposit.Address, error) {
+	response, err := ok.GetCurrencyDepositAddress(ctx, c.String())
+	if err != nil {
+		return nil, err
+	}
+
+	for x := range response {
+		if accountID == response[x].Address && (strings.EqualFold(response[x].Chain, chain) || strings.HasPrefix(response[x].Chain, c.String()+"-"+chain)) {
+			return &deposit.Address{
+				Address: response[x].Address,
+				Tag:     response[x].Tag,
+				Chain:   response[x].Chain,
+			}, nil
+		}
+	}
+	if len(response) > 0 {
+		return &deposit.Address{
+			Address: response[0].Address,
+			Tag:     response[0].Tag,
+			Chain:   response[0].Chain,
+		}, nil
+	}
+	return nil, errDepositAddressNotFound
+}
+
+// WithdrawCryptocurrencyFunds returns a withdrawal ID when a withdrawal is submitted
+func (ok *Okx) WithdrawCryptocurrencyFunds(ctx context.Context, withdrawRequest *withdraw.Request) (*withdraw.ExchangeResponse, error) {
+	if err := withdrawRequest.Validate(); err != nil {
+		return nil, err
+	}
+	var input WithdrawalInput
+	input.ChainName = withdrawRequest.Crypto.Chain
+	input.Amount = withdrawRequest.Amount
+	input.Currency = withdrawRequest.Currency.String()
+	input.ToAddress = withdrawRequest.Crypto.Address
+	input.TransactionFee = withdrawRequest.Crypto.FeeAmount
+	input.WithdrawalDestination = "3"
+	resp, err := ok.Withdrawal(ctx, &input)
+	if err != nil {
+		return nil, err
+	}
+	return &withdraw.ExchangeResponse{
+		ID: resp.WithdrawalID,
+	}, nil
+}
+
+// WithdrawFiatFunds returns a withdrawal ID when a withdrawal is
+// submitted
+func (ok *Okx) WithdrawFiatFunds(ctx context.Context, withdrawRequest *withdraw.Request) (*withdraw.ExchangeResponse, error) {
+	return nil, common.ErrNotYetImplemented
+}
+
+// WithdrawFiatFundsToInternationalBank returns a withdrawal ID when a withdrawal is submitted
+func (ok *Okx) WithdrawFiatFundsToInternationalBank(ctx context.Context, withdrawRequest *withdraw.Request) (*withdraw.ExchangeResponse, error) {
+	return nil, common.ErrNotYetImplemented
+}
+
+// GetActiveOrders retrieves any orders that are active/open
+func (ok *Okx) GetActiveOrders(ctx context.Context, req *order.GetOrdersRequest) ([]order.Detail, error) {
+	if err := req.Validate(); err != nil {
+		return nil, err
+	}
+	if len(req.Pairs) == 0 || len(req.Pairs) >= 40 {
+		req.Pairs = append(req.Pairs, currency.EMPTYPAIR)
+	}
```

Comment:
This seems quite strange. Could you explain this to me? If req.Pairs is greater than 40, why would it benefit from the addition of an empty pair?

In the for loop below, if a currency does not equal a `req.Pair` you `continue`, so it will never match. 
In the scenario where `req.Pairs == 0`, all it does is go through the loop `for x:= range response` but does not do anything.